go/session_test.go

```diff
@@ -0,0 +1,61 @@
+package testGoSDK
+
+import (
+	"testing"
+
+	xenapi "github.com/xenserver/xenserver-samples/go/goSDK"
+)
+
+func TestHTTPSConnection(t *testing.T) {
+	if *CA_CERT_PATH_FLAG == "" {
+		t.Log("CA certificate is not provided, skipping https connetcion test")
+		t.Fail()
+		return
+	}
+	// Test HTTPS connection without certificate verification
+	session1 := xenapi.NewSession(&xenapi.ClientOpts{
+		URL: "https://" + *IP_FLAG,
+	})
+
+	_, err := session1.LoginWithPassword(*USERNAME_FLAG, *PASSWORD_FLAG, "1.0", "Go sdk samples")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	err = session1.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	// Test HTTPS connection with server certificate verification
+	if session.APIVersion >= xenapi.APIVersion2_21 {
+		/* the the CA cert in yangtze server missing fields which will cause x509 error:
```

Comment:
"the the CA cert in yangtze server missing fields which..." => "the CA cert in yangtze servers is missing fields, which"